lib/api/src/index.tsx

```diff
@@ -0,0 +1,255 @@
+import initShortcuts, {
+  SubState as ShortcutsSubState,
+  SubAPI as ShortcutsAPI,
+} from './modules/shortcuts';
+import initURL, { QueryParams } from './modules/url';
+import initVersions, { SubState as VersionsSubState } from './modules/versions';
+
+const ManagerContext = createContext({ api: undefined, state: getInitialState({}) });
+
+const { STORY_CHANGED, SET_STORIES, SELECT_STORY } = Events;
+
+export type Module = StoreData & RouterData & ProviderData;
+
+export type State = Other &
+  LayoutSubState &
+  StoriesSubState &
+  NotificationState &
+  VersionsSubState &
+  RouterData &
+  ShortcutsSubState;
+
+export type API = AddonsAPI &
+  ChannelAPI &
+  StoriesAPI &
+  LayoutAPI &
+  NotificationAPI &
+  ShortcutsAPI &
+  OtherAPI;
+
+interface OtherAPI {
+  [key: string]: any;
+}
+interface Other {
+  customQueryParams: QueryParams;
+
+  notifications: Notification[];
+
+  [key: string]: any;
+}
+
+export interface Combo {
+  api: API;
+  state: State;
+}
+
+interface Provider {
+  handleAPI(api: API): void;
+  renderPreview(): void;
+  getElements(type: Types): Collection;
+  [key: string]: any;
+}
+
+interface ProviderData {
+  provider: Provider;
+}
+
+interface StoreData {
+  store: Store;
+}
+
+interface Children {
+  children: React.Component | ((props: Combo) => React.Component);
+}
+
+type Props = Children & RouterData & ProviderData;
+
+class ManagerProvider extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { provider, location, path, viewMode, storyId, navigate } = props;
+
+    const store = new Store({
+      getState: () => this.state,
+      setState: (a: any, b: any) => this.setState(a, b),
+    });
+
+    // Initialize the state to be the initial (persisted) state of the store.
+    // This gives the modules the chance to read the persisted state, apply their defaults
+    // and override if necessary
+    this.state = store.getInitialState(getInitialState({}));
+
+    const apiData = {
+      navigate,
+      store,
+      provider,
+      location,
+      path,
+      viewMode,
+      storyId,
+    };
+
+    this.modules = [
+      initChannel,
+      initAddons,
+      initLayout,
+      initNotifications,
+      initShortcuts,
+      initStories,
+      initURL,
+      initVersions,
+    ].map(initModule => initModule(apiData));
+
+    // Create our initial state by combining the initial state of all modules, then overlaying any saved state
+    const state = getInitialState(...this.modules.map(m => m.state));
+
+    // Get our API by combining the APIs exported by each module
+    const combo = Object.assign({ navigate }, ...this.modules.map(m => m.api));
+
+    const api = initProviderApi({ provider, store, api: combo });
+
+    api.on(STORY_CHANGED, (id: string) => {
+      const options = api.getParameters(id, 'options');
+
+      api.setOptions(options);
+    });
+
+    api.on(SET_STORIES, (data: any) => {
+      api.setStories(data.stories);
```

Comment:
You are accessing `data.stories` even though you use `data: any` 
So you at least know at this point you have `data: { stories: SomeStoryType }`

Reply:
Good catch this object is well known and I can create a type for it, I may already have it.